### DISCUSSION TITLE
Restore pill upon undo using creatable in react-select

### DISCUSSION THREAD

**Discussion by karanshah229** [JedWatson/react-select#5024] [Ideas]
When you create a pill using `creatable` in `react-select`, you have the option to delete it as well.

So, after deleting it using `backspace`, if one tries to undo, the creatable will not be restored.

This is counter-intuitive, since if you delete a word in input tag and undo, it will be restored.

I am open to work-arounds if possible and would greatly appreciate any help 🙏🏽

**Comment by Rall3n** [comment]
I believe this is something that should be handled by the developing user.
This way you would have complete control over how a history is being saved.

[Here is an example](https://codesandbox.io/s/react-select-with-history-ugdpy) using the library `react-state-history-tree`.
